python/amici/sbml_import.py

```diff
@@ -522,6 +519,29 @@ def processCompartments(self):
                comp, vol
             )
 
+    def processRateRules(self):
+        """Adapt SBML model to cover Rate Rules as well.
+
+        Arguments:
+
+        Returns:
+
+        Raises:
+
+        """
+
+        for rule in self.sbml.getModel().getListOfRules():
+            if rule.isRate():
+
+                # set constant input reactions with kinetic law
+                # equal to the right hand site of the Rate Rule
+                r = self.sbml.createReaction()
+                r.setId('d_dt_' + rule.getVariable())
+                r.setName('d_dt_' + rule.getVariable())
+                r.createProduct().setSpecies(rule.getVariable())
```

Comment:
The rule variable can also be a parameter, this would have to be treated differently (see SBML specs 4.9.4).

Please run the SBML test suite with these changes (`scripts/run-SBMLTestsuite.sh`). This should include a couple of cases for RateRules.

I am not sure if we want to alter the SBML object here, or not rather just add these reactions to the symbols of SBMLImporter (@FFroehlich ?)

Reply:
Thank you for your pointing this out!

If it is to a parameter, should we create a new species, that is "equal to the parameter", and apply the rate rule to the new species? (And also later, what if we introduce variable compartment size and the rate rule acts on the compartment? That we need to keep in mind for later :) )

Reply:
We do not want to ever allow rate rules for parameters as respective sensitivities are not well defined.

I would actually say that flattening rate rules to reactions is a good thing as it simplifies downstream processing (it will give rise to a stoichiometric matrix + flux vector etc that we can use later on)

Reply:
@FFroehlich @dweindl  Is it the right way to go, to modify the libsbml model? Or do we want to modify the sympy-part of SBML_import? (In that case I would have to take a deeper dive into the code's structure and would for sure need some more week(s?) more...)

Reply:
I think its fine to modify the SBML, however I think this functionality should not be part of `sbml_import.py`. I would rather have this implemented as general purpose 'flattener' that acts on an sbml object, which is passed to import_sbml as preprocessing, similar to what we have for observables/constants etc.

Reply:
Okay then. As Fabi said, then this should be a free function.